tasks/check-e2e-irrelevant-pr/0.1/check-e2e-irrelevant-pr.yaml

```diff
@@ -0,0 +1,118 @@
+      workingDir: /workspace
+      env:
+        - name: GIT_ORGANIZATION
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.labels['pac.test.appstudio.openshift.io/url-org']
+        - name: GIT_REPOSITORY
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.labels['pac.test.appstudio.openshift.io/url-repository']
+        - name: PR_NUMBER
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.labels['pac.test.appstudio.openshift.io/pull-request']
+      script: |
+        #!/usr/bin/env bash
+        set -euo pipefail
+
+        # Function to handle exit and set result
+        exitfunc() {
+          local err=$1
+          local line=$2
+          local command="$3"
+          if [ "$err" -eq 0 ]; then
+            echo -n "true" | tee "$(results.result.path)"
+            exit 0
+          elif [ "$err" -eq 99 ]; then
+            echo -n "false" | tee "$(results.result.path)"
+            exit 0
+          else
+            echo "$0: ERROR $command failed at line $line - exited with status $err"
+            echo -n "true" | tee "$(results.result.path)"
+            exit "$err"
+          fi
+        }
+        trap 'exitfunc $? $LINENO "$BASH_COMMAND"' EXIT
+
+        # Setup workspace
+        WORK_DIR=".tmp_pr_check"
+        rm -rf "$WORK_DIR"
+        mkdir "$WORK_DIR"
+        cd "$WORK_DIR"
+
+        # Clone and checkout PR
+        if ! git clone --quiet "https://github.com/${GIT_ORGANIZATION}/${GIT_REPOSITORY}" "${GIT_REPOSITORY}"; then
+          echo "Failed to clone repository"
+          exit 1
+        fi
+        cd "${GIT_REPOSITORY}"
+        if ! git fetch origin pull/"$PR_NUMBER"/head:pr_"$PR_NUMBER" > /dev/null 2>&1; then
+          echo "Failed to fetch PR"
+          cd ../..
+          rm -rf "$WORK_DIR"
+          exit 1
+        fi
+        if ! git checkout pr_"$PR_NUMBER" > /dev/null 2>&1; then
+          echo "Failed to checkout PR branch"
+          cd ../..
+          rm -rf "$WORK_DIR"
+          exit 1
+        fi
+
+        # Get changed files
+        changed_files=$(git diff --name-only origin/"$(params.base-branch)"...HEAD)
+
+        irrelevant_pattern='(^|.*/)(\.gitignore|PROJECT|LICENSE|\.sonarcloud\.properties|\.gitlint|\.dockerignore)$'
+
+        if [[ -z "$changed_files" ]]; then
+          echo "No files changed."
+          cd ../..
+          rm -rf "$WORK_DIR"
+          exit 0
+        fi
+
+        all_irrelevant=true
+
+        while IFS= read -r file; do
+          # Consider as irrelevant if matches any of the following:
+          if [[ "$file" =~ ^\.github/.* ]] || [[ "$file" =~ ^.*\.md$ ]] || [[ "$file" =~ $irrelevant_pattern ]]; then
+            continue
+          else
+            echo "$file --> relevant"
+            all_irrelevant=false
+            break
+          fi
+        done <<< "$changed_files"
+
+        cd ../..
+        rm -rf "$WORK_DIR"
```

Comment:
not sure if we need to do the cleanup since the task is running in a container with ephemeral storage, wdyt?

Reply:
OK. Removed them.